node/impl/full/eth_trace.go

```diff
@@ -0,0 +1,339 @@
+			Input: nil,
+			Value: ethtypes.EthBigInt(et.Msg.Value),
+
+			FilecoinFrom:    et.Msg.From,
+			FilecoinTo:      et.Msg.To,
+			FilecoinMethod:  et.Msg.Method,
+			FilecoinCodeCid: et.Msg.CodeCid,
+		},
+		Result: ethtypes.EthTraceResult{
+			GasUsed: ethtypes.EthUint64(et.SumGas().TotalGas),
+			Output:  nil,
+		},
+		Subtraces:    len(et.Subcalls),
+		TraceAddress: addr,
+
+		Parent: parent,
+	}
+
+	trace.SetCallType("call")
+
+	if et.Msg.Method == builtin.MethodsEVM.InvokeContract {
+		log.Debugf("found InvokeContract call at height: %d", height)
+
+		// TODO: ignore return errors since actors can send gibberish and we don't want
+		// to fail the whole trace in that case
+		trace.Action.Input, err = decodePayload(et.Msg.Params, et.Msg.ParamsCodec)
+		if err != nil {
+			return xerrors.Errorf("buildTraces: %w", err)
+		}
+		trace.Result.Output, err = decodePayload(et.MsgRct.Return, et.MsgRct.ReturnCodec)
+		if err != nil {
+			return xerrors.Errorf("buildTraces: %w", err)
+		}
+	} else if et.Msg.To == builtin.EthereumAddressManagerActorAddr &&
+		et.Msg.Method == builtin.MethodsEAM.CreateExternal {
+		log.Debugf("found CreateExternal call at height: %d", height)
+		trace.Action.Input, err = decodePayload(et.Msg.Params, et.Msg.ParamsCodec)
+		if err != nil {
+			return xerrors.Errorf("buildTraces: %w", err)
+		}
+
+		if et.MsgRct.ExitCode.IsSuccess() {
+			// ignore return value
+			trace.Result.Output = nil
+		} else {
+			// return value is the error message
+			trace.Result.Output, err = decodePayload(et.MsgRct.Return, et.MsgRct.ReturnCodec)
+			if err != nil {
+				return xerrors.Errorf("buildTraces: %w", err)
+			}
+		}
+
+		// treat this as a contract creation
+		trace.SetCallType("create")
+	} else {
+		trace.Action.Input, err = handleFilecoinMethodInput(et.Msg.Method, et.Msg.ParamsCodec, et.Msg.Params)
+		if err != nil {
+			return xerrors.Errorf("buildTraces: %w", err)
+		}
+		trace.Result.Output, err = handleFilecoinMethodOutput(et.MsgRct.ExitCode, et.MsgRct.ReturnCodec, et.MsgRct.Return)
+		if err != nil {
+			return xerrors.Errorf("buildTraces: %w", err)
+		}
+	}
+
+	// TODO: is it OK to check this here or is this only specific to certain edge case (evm to evm)?
+	if et.Msg.ReadOnly {
+		trace.SetCallType("staticcall")
+	}
+
+	// there are several edge cases thar require special handling when displaying the traces
+	if parent != nil {
+		// Handle Native actor creation
+		//
+		// Actor A calls to the init actor on method 2 and The init actor creates the target actor B then calls it on method 1
+		if parent.Action.FilecoinTo == builtin.InitActorAddr &&
+			parent.Action.FilecoinMethod == builtin.MethodsInit.Exec &&
+			et.Msg.Method == builtin.MethodConstructor {
+			log.Debugf("Native actor creation! method:%d, code:%s, height:%d", et.Msg.Method, et.Msg.CodeCid.String(), height)
+			parent.SetCallType("create")
+			parent.Action.To = et.Msg.To.String()
+			parent.Action.Input = []byte{0x0, 0x0, 0x0, 0xFE}
```

Comment:
This reads as STOP STOP STOP INVALID. I don't think we need the padding, but I'm not sure.

Reply:
Ah, I probably have confused my byte-2-byte testing which I did against the builtinactors version of handleFilecoinMethodInput/handle_filecoin_method_input to this Go versions I have here, and not specifically to what to return here!